### PR TITLE
Xml app branch

### PR DESCRIPTION
**Pull request by Tejjj**
[xml_application.webm](https://user-images.githubusercontent.com/20522801/236686266-e9bffd37-79ec-4eb1-9017-7f413c0a3796.webm)

### CODE REVIEW

File: xml-app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+plugins {
+    id("com.android.application")
+    id("org.jetbrains.kotlin.android")
+    id("dagger.hilt.android.plugin")
+    id("androidx.navigation.safeargs.kotlin")
+    kotlin("kapt")
+}
+
+android {
+    namespace = "com.example.xml_app"
```
Comment (hadiyarajesh) [inline xml-app/build.gradle.kts:26]: Can you set the package name to com.hadiyarajesh.xml_app? (For whole app, not just in build file)

File: xml-app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        release {
+            isMinifyEnabled = false
+            proguardFiles(
+                getDefaultProguardFile("proguard-android-optimize.txt"),
+                "proguard-rules.pro"
+            )
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (hadiyarajesh) [inline xml-app/build.gradle.kts:49]: Compile to at least java 11, preferably java 17

File: xml-app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+
+    buildFeatures {
+        viewBinding = true
```
Comment (hadiyarajesh) [inline xml-app/build.gradle.kts:57]: why are we using both data and view binding? isn't one is sufficient?

File: xml-app/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+   /* implementation("androidx.core:core-ktx:1.9.0")
+    implementation("androidx.appcompat:appcompat:1.6.1")
+    implementation("com.google.android.material:material:1.8.0")
+    implementation("androidx.constraintlayout:constraintlayout:2.1.4")
+    testImplementation("junit:junit:4.13.2")
+    androidTestImplementation("androidx.test.ext:junit:1.1.5")
+    androidTestImplementation("androidx.test.espresso:espresso-core:3.5.1")*/
+
+    implementation("androidx.legacy:legacy-support-v4:1.0.0")
```
Comment (hadiyarajesh) [inline xml-app/build.gradle.kts:84]: why do we need legacy support dependency?

File: xml-app/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    // LiveData
+    implementation("androidx.lifecycle:lifecycle-livedata-ktx:2.6.1")
```
Comment (hadiyarajesh) [inline xml-app/build.gradle.kts:87]: can you make it to state flow instead of live data?

File: xml-app/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    // ViewModel
+    implementation("androidx.lifecycle:lifecycle-viewmodel-ktx:2.6.1")
+
+    // Kotlin-navigation
+    implementation("androidx.navigation:navigation-fragment-ktx:2.5.3")
+    implementation("androidx.navigation:navigation-ui-ktx:2.5.3")
+
+    val nav_version = "2.5.3"
```
Comment (hadiyarajesh) [inline xml-app/build.gradle.kts:96]: define these variables in `LibVersion` object

File: xml-app/src/main/java/com/example/xml_app/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import androidx.navigation.ui.navigateUp
+import androidx.navigation.ui.setupActionBarWithNavController
+import androidx.navigation.ui.setupWithNavController
+import com.example.xml_app.databinding.ActivityMainBinding
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var lbinding: ActivityMainBinding
```
Comment (hadiyarajesh) [inline xml-app/src/main/java/com/example/xml_app/MainActivity.kt:34]: can we use the suitable variable name here (and in whole app)?

File: xml-app/src/main/java/com/example/xml_app/di/NetworkModule.kt [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            .baseUrl(API_BASE_URL)
+            .client(okHttpClient)
+            .addCallAdapterFactory(FlowerCallAdapterFactory.create())
+            .addConverterFactory(MoshiConverterFactory.create())
+            .build()
+    }
+
+    @Provides
+    @Singleton
+    fun provideQuoteApi(retrofit: Retrofit): ImageApi = retrofit.create(ImageApi::class.java)
```
Comment (hadiyarajesh) [inline xml-app/src/main/java/com/example/xml_app/di/NetworkModule.kt:52]: please rename this function

File: xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                        is UiState.Success -> {
+                            profileListState.data.let {
+                                profileAdapter.submitList(it)
+                            }
+                        }
+
+                        UiState.Empty -> {
+
+                        }
+                        is UiState.Error -> TODO()
```
Comment (hadiyarajesh) [inline xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt:79]: Please handle /9or remove) this TODO

File: xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    private fun fetchData() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.images.collectLatest { profileListState ->
+                    when(profileListState) {
+                        is UiState.Loading -> {
+                            // Show progress bar..
```
Comment (hadiyarajesh) [inline xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt:67]: No UI here?

**Review by hadiyarajesh — CHANGES_REQUESTED** [review]

File: xml-app/build.gradle.kts (hunk 2 above)
Comment (Tejjj) [reply xml-app/build.gradle.kts:49]: The Safe Args plugin, which is part of the Android Jetpack Navigation component, requires at least Java 8 (Java SE 8) or higher to work properly.

File: xml-app/build.gradle.kts (hunk 3 above)
Comment (Tejjj) [reply xml-app/build.gradle.kts:57]: yes , will submit by removing the viewBinding .

File: xml-app/build.gradle.kts (hunk 4 above)
Comment (Tejjj) [reply xml-app/build.gradle.kts:84]: will remove this .

File: xml-app/build.gradle.kts (hunk 5 above)
Comment (Tejjj) [reply xml-app/build.gradle.kts:87]: Will remove this wrong dependency, I am using  StateFlow in the application.

File: xml-app/src/main/java/com/example/xml_app/di/NetworkModule.kt (hunk 8 above)
Comment (Tejjj) [reply xml-app/src/main/java/com/example/xml_app/di/NetworkModule.kt:52]: ok

File: xml-app/build.gradle.kts (hunk 6 above)
Comment (Tejjj) [reply xml-app/build.gradle.kts:96]: ok

File: xml-app/src/main/java/com/example/xml_app/MainActivity.kt (hunk 7 above)
Comment (Tejjj) [reply xml-app/src/main/java/com/example/xml_app/MainActivity.kt:34]: included these changes in the new commit

File: xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt (hunk 10 above)
Comment (Tejjj) [reply xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt:67]: will submit in new commit

File: xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt (hunk 9 above)
Comment (Tejjj) [reply xml-app/src/main/java/com/example/xml_app/fragment/home/ProfileListFragment.kt:79]: ok

**Pull request merged**